### PR TITLE
Don't crash when #[dom_struct] is applied on an empty struct

### PR DESCRIPTION
**Pull request by Manishearth**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13425)

### CODE REVIEW

File: components/plugins/reflector.rs [hunk 1]
```diff
@@ -12,6 +12,9 @@ pub fn expand_reflector(cx: &mut ExtCtxt, span: Span, _: &MetaItem, annotatable:
                         push: &mut FnMut(Annotatable)) {
     if let &Annotatable::Item(ref item) = annotatable {
         if let ItemKind::Struct(ref def, _) = item.node {
+            if def.fields().len()== 0 {
```
Comment (Ms2ger) [inline components/plugins/reflector.rs]: Space before `==`

File: components/plugins/reflector.rs [hunk 2]
```diff
@@ -12,6 +12,9 @@ pub fn expand_reflector(cx: &mut ExtCtxt, span: Span, _: &MetaItem, annotatable:
...
+                cx.span_err(span, "#[dom_struct] should have a reflector or parent dom struct as its first field");
```
Comment (Ms2ger) [inline components/plugins/reflector.rs]: Do we need to return here?

**Review by Ms2ger — CHANGES_REQUESTED** [review]

**Comment by Manishearth** [comment]
Fixed both

-Manish Goregaokar

File: components/plugins/reflector.rs (hunk 1 above)
Comment (emilio) [inline components/plugins/reflector.rs]: Or even better, `def.fields.is_empty()`.

**Comment by emilio** [comment]
r=me,Ms2ger with or without the `is_empty` thing.

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**